scripts/lib/CIME/tests/test_compare_test_results.py

```diff
@@ -0,0 +1,98 @@
+#/usr/bin/env python
+
+"""
+This module contains unit tests for compare_test_results
+"""
+
+import unittest
```

Comment:
@jasonb5 , thanks for adding a test! We have been putting most of our testing in scripts/tests/scripts_regression_tests.py. You can put this code there or keep it here and just make sure scripts_regression_tests.py calls this test.

Reply:
@jgfouca that's not true: we have a number of unit tests in the directory where @jasonb5 added this, and if things are set up right, they should all be invoked from A_RunUnitTests. We had a discussion a few years ago in which a few of us agreed that it would probably be better to put new unit tests in their own module there rather than adding to the already-huge scripts_regression_tests. Which is all to say: I like how @jasonb5 set up this test.

Reply:
@jgfouca @billsacks I'll keep the test were it is and double check that it's being excercised with A_RunUnitTests.

Reply:
Thanks guys, I remember that discussion.

Reply:
Ok double checked the tests are picked up by A_RunUnitTests. This can be resolved and i'll move the PR out of draft.